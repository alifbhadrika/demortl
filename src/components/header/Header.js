import React from "react";

function Header({ text }) {
  return (
    <div className="Header" style={{ marginBottom: 40 }}>
      {text}
    </div>
  );
}

export default Header;
