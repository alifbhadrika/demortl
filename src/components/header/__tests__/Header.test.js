import React from "react";
import Header from "../Header";
import { render } from "@testing-library/react";

describe("Tests for Header component", () => {
  it("snapshot header component", () => {
    const mockText = "This is just for the sake of the test";
    const { asFragment } = render(<Header text={mockText} />);
    expect(asFragment(<Header text={mockText} />)).toMatchSnapshot();
  });

  it("should render header component with the right text", () => {
    const mockText = "This is just for the sake of the test";
    const { getByText } = render(<Header text={mockText} />);
    expect(getByText(mockText)).not.toBeNull();
  });

  it("alternative way using toBeInTheDocument jest-dom uitility library", () => {
    const mockText = "This is just for the sake of the test";
    const { getByText } = render(<Header text={mockText} />);
    expect(getByText(mockText)).toBeInTheDocument();
  });
});
