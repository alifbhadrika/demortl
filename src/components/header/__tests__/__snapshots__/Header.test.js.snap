// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Header component snapshot header component 1`] = `
<DocumentFragment>
  <div
    class="Header"
    style="margin-bottom: 40px;"
  >
    This is just for the sake of the test
  </div>
</DocumentFragment>
`;
